### PR TITLE
Add full text search support for mysql and postgres db

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
Add full-text search support for channel names.  Using full text search support for mysql and postgres.

Requires adding full-text search index to the Channels table for (Name, DisplayName).

#### mysql
``` mysql
ALTER TABLE Channels
ADD FULLTEXT(Name, DisplayName);
```

#### postgres
``` postgres
Not 100% sure if postgres requires any additional changes.
```

#### Checklist
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by dmeza** [comment]
@miguelespinoza please help review, there are some tests that are not passing:
```
        --- FAIL: TestChannelStore/MySQL/SearchMore (0.02s)
        	channel_store.go:2247: wrong channel returned
        --- FAIL: TestChannelStore/MySQL/SearchInTeam (0.02s)
        	channel_store.go:2427: wrong channel returned
        --- FAIL: TestChannelStore/PostgreSQL/SearchMore (0.01s)
        	channel_store.go:2216: should return 2 channels, not including private channel
        --- FAIL: TestChannelStore/PostgreSQL/SearchInTeam (0.01s)
        	channel_store.go:2396: should return 2 channels, not including private channel
```

**Comment by miguelespinoza** [comment]
Some thoughts before I make changes to the test.

Adding fuzzy text search adds a bit of fragility to the test.  We are not searching for exact matches anymore.  We are using the FULLTEXT search capabilities from `MySQL` and `Postgres`.  This returns back the closest values based on the string it's matching.

I realized after some testing that it was returning zero values for this searchQuery using `IN BOOLEAN MODE`
changing to `IN NATURAL LANGUAGE MODE` did the trick.

Also, upon some research, I found that MySQL only indexes fields with a minimum length based on this config value `ft_min_word_len ` can be found in `/etc/mysql/my.cnf`
https://www.electrictoolbox.com/mysql-full-text-index-word-length/

I would like to avoid this solution since it would require to restart the database and recreate the index table.  This is an important bit of info since we are dealing with Channel names and there can be some names that may be below 4 characters in length (the default value for MySQL in the docker containers).  This could affect the search results for certain names

Here's more information about fine tuning fulltext for mysql: https://dev.mysql.com/doc/refman/5.7/en/fulltext-fine-tuning.html

In conclusion, I want you all to be aware of the implications of these changes.  I can then remove certain test since we are not going to be able to rely on them moving forward because of the nature of this feature

**Comment by miguelespinoza** [comment]
Also, I decided to add Purpose as a searchable field when looking up Channels.  This was a requested feature which would enhance the searching experience

**Comment by miguelespinoza** [comment]
Would adding the Header field also be beneficial? 
If we can account for all searchable fields now we won't have to worry about adding them later down the road and have to worry about re-indexing the table

**Comment by miguelespinoza** [comment]
Hmm, looks like another PR has FULL_TEXT Support. I'll close this and create a new PR for adding Purpose as a searchable field

**Comment by esethna** [comment]
Closing as per comment ^

PR for purpose field: https://github.com/mattermost/mattermost-server/pull/8067
